cate/ds/esa_cci_odp.py

```diff
@@ -278,19 +286,45 @@ def _get_linked_content_from_descxml_elem(descxml: etree.XML, paths: List[str])
             return _get_element_content(descxml_elem, paths[2])
 
 
+async def get_response(session: aiohttp.ClientSession, url: str) -> Optional[aiohttp.ClientResponse]:
+    response = None
+    retry_backoff_max = _RETRY_BACKOFF_MAX
+    for i in range(_NUM_RETRIES):
+        resp = await session.request(method='GET', url=url)
+        if resp.status == 200:
+            return resp
+        elif 500 <= resp.status < 600:
+            # Retry (immediately) on 5xx errors
+            continue
+        elif resp.status == 429:
+            # Retry after 'Retry-After' with exponential backoff
+            retry_min = int(response.headers.get('Retry-After', '100'))
+            retry_backoff = random.random() * retry_backoff_max
+            retry_total = retry_min + retry_backoff
+            retry_message = f'Error 429: Too Many Requests. ' \
+                            f'Attempt {i + 1} of {_NUM_RETRIES} to retry after ' \
+                            f'{"%.2f" % retry_min} + {"%.2f" % retry_backoff} = {"%.2f" % retry_total} ms...'
+            warnings.warn(retry_message)
+            time.sleep(retry_total / 1000.0)
+            retry_backoff_max *= _RETRY_BACKOFF_BASE
+        else:
+            break
+    return None
```

Comment:
If I see it right, in the end, the actual error information is lost and we'll never know what went wrong. If this is true, please return the HTTP error and status message so users and developers are informed about what went wrong.